lib/sprockets/sass/registration.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+module Sprockets
+  module Sass
+    # class useful for registering engines, tranformers, preprocessors and conpressors for sprockets
+    # depending on the version of sprockets
+    class Registration
+      attr_reader :klass
+
+      AVAILABLE_VERSIONS = [4, 3, 2].freeze
+
+      DEFAULT_ACTION = { min_comparison: '>=', min: 2, max_comparison: '<', max: 3, action: :register_sprockets_legacy }.freeze
+
+      ACTIONS = [
+        { min_comparison: '>=', min: 3, max_comparison: '<', max: 4, action: :register_sprockets_v3 },
+        #    { min_comparison: '>=' , min: 4,  action: :register_sprockets_v4 }
+      ].freeze
+
+      attr_reader :klass, :sprockets_version, :registration_instance, :action_details, :version_selected
+
+      def initialize(klass)
+        @klass = klass
+        @sprockets_version = Sprockets::Sass::Utils.version_of_sprockets
+        @registration_instance = self
+        @action_details = action_to_be_executed
+        @version_selected = @action_details[:min]
+      end
+
+      def run
+        require_libraries
+        send(@action_details[:action])
+      end
+
+      def require_libraries
+        AVAILABLE_VERSIONS.each do |version|
```

Comment:
Shouldn't this only require the the libraries for the current version of sprockets? If we're requiring all of them, we could just explicitly list the require statements